packages/react/src/hooks/useAuth.test.tsx

```diff
@@ -0,0 +1,49 @@
+import { render, screen, userEvent } from '@clerk/shared/testUtils';
+import React, { useEffect, useState } from 'react';
+
+import { useAuth } from './useAuth';
+import { ClerkProvider } from '../contexts/ClerkProvider';
+
+const withFakeClerk = Component => {
+  return props => (
+    <ClerkProvider frontendApi='https://example.com'>
+      <Component {...props} />
+    </ClerkProvider>
+  );
+};
+
+describe('useAuth()', () => {
+  it('auth is stable between re-renders', async () => {
+    let renderCount = 0;
+    const MyDemo = () => {
```

Comment:
🔧  An alternative and more expressive way to test this is to rerender the component via prop updates as shown at https://testing-library.com/docs/example-update-props/

Reply:
Wow I didn't know that one, that seems very interesting! I'll definitely add a similar thing to [my own React testing library, thanks for the tip!](https://react-test.dev/). Let me see if I can get it to work here as well, but seems like it should be easy.

Reply:
Fixed as suggested, thanks! (also [added it to my testing library](https://react-test.dev/documentation#props) 🥳)